### ISSUE TITLE
SSDBComboBox.updateOption() not always updating combo editor text

### ISSUE THREAD

**Issue by bpangburn** [bpangburn/swingset#85]
Demo->Example 4 With Helper:

Added a focus lost listener to Part Name to update the combo navigator via a call to SSDBComboBox.updateOption() if the Part Name changes. Working as expected for the first record where part name is edited, but not on other rows. Sometimes a navigation takes place bringing the user back to the first row.

Steps to replicate:
1. Open Example 4 Using Helper
2. Change Part Name from "Nut" to "Nutty" and tab to next field. Navigator updates.
3. Move to 2nd record.
4. Change Part Name from "Bolt" to "Bolty" and tab to next field. Navigator fails to update.
5. Navigate to record 3 then back to 2. Update is there.

Doing a refresh on the data navigator resets things.

**Comment by bpangburn** [comment]
@errael was able to fix SSDBComboBox.updateOption() with the following modifications.

```
	public boolean updateOption(final Long _mapping, final String _option) {

		boolean result = false;

		try (Model.Remodel remodel = optionModel.getRemodel()) {
			final int index = remodel.getMappings().indexOf(_mapping);
			if (index >= 0) {
				boolean isSelectedItem = Objects.equals(_mapping, getSelectedMapping());
				remodel.setOption(index, _option);
				result = true;
				if (isSelectedItem) {
					// Modifying the underlying list item that corresponds to the
					// current selection; strict glazed may change the selection.
					// Select the modified item so the same mapping is selected.
					SSListItem item = remodel.get(index);
					setSelectedItem(item);
				}
// TODO Confirm that eventList is not reordered by GlazedLists code.
			}
		} catch (final Exception e) {
			logger.error(getColumnForLog() + ": Exception.", e);
		}
		
		if (!result) {
			logger.warn(() -> String.format("%s: Unable to update Mapping of [%s] with Option of '%s'.",
				getColumnForLog(), _mapping, _option));
		}

		return result;
	}
```
@errael also opened an issue in GlazedLists with further insights:
https://github.com/glazedlists/glazedlists/issues/702

**Comment by bpangburn** [comment]
Fixed in swingset-4.0.4
